src/components/input/Button/Button.styles.js

```diff
@@ -1,65 +1,82 @@
 import styled, { css } from 'styled-components';
 
 import { buttonColors } from '../../../base/color/colorVariables';
-import fontSizes from '../../../base/typography/fontSizes';
+import { interfaceTypography } from '../../../base/typography/typographyVariables';
+import spacerSizes from '../../../base/spacing';
 
-export const buttonModes = {
+
+export const buttonProminences = {
   primary: css`
-    background-color: ${buttonColors.buttonPrimaryBackground};
-    border: 1px solid ${buttonColors.buttonPrimaryBorder};
-    color: ${buttonColors.buttonPrimaryText};
+    background-color: ${setButtonColour(intent)};
+    border: none;
+    border-radius: 10px;
+    color: ${buttonColors.primaryText};
 
     &:hover {
-      background-color: ${buttonColors.buttonPrimaryBackgroundHover};
+      background-color: ${buttonColors.hover};
     }
   `,
-  secondary: css`
-    background-color: ${buttonColors.buttonSecondaryBackground};
-    border: 1px solid ${buttonColors.buttonSecondaryBorder};
-    color: ${buttonColors.buttonSecondaryText};
+  default: css`
+    background: none;
+    border: 2px solid ${buttonColors.default};
+    border-radius: 10px;
+    color: ${buttonColors.default};
 
     &:hover {
-      background-color: ${buttonColors.buttonSecondaryBackgroundHover};
+      border-color: ${buttonColors.hover};
+      color: ${buttonColors.hover};
     }
   `,
-  danger: css`
-    background-color: ${buttonColors.buttonDangerBackground};
-    border: 1px solid ${buttonColors.buttonDangerBorder};
-    color: ${buttonColors.buttonDangerText};
+  minimal: css`
+    background: none;
+    border: none;
+    border-radius: 10px;
+    color: ${buttonColors.default};
 
     &:hover {
-      background-color: ${buttonColors.buttonDangerBackgroundHover};
+      color: ${buttonColors.hover};
     }
   `,
-  success: css`
-    background-color: ${buttonColors.buttonSuccessBackground};
-    border: 1px solid ${buttonColors.buttonSuccessBorder};
-    color: ${buttonColors.buttonSuccessText};
+};
 
-    &:hover {
-      background-color: ${buttonColors.buttonSuccessBackgroundHover};
-    }
+// success, warning, danger, none
+export const buttonIntents = {
+  success: css`
+    
+    }, 
   `,
 };
 
-export const buttonSizes = {
-  small: css`
-    ${fontSizes.size0};
-    padding: 8px 11px;
+// We're going to try to colour the buttons based on state or intent
+setButtonColour(intent) {
```

Comment:
Functions need to be prefixed with `function` to work 😉. So `function setButtonColour() {}` or `const setButtonColour = () => {}`. I'd suggest:

```suggestion
export const setButtonColour = (intent) => {
```

Reply:
Thanks! I changed this but it still doesn't solve my problem, which I've articulated in more detail below.